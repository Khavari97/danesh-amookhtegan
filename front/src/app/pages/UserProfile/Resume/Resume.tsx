import React, {Component} from 'react';
import Style from "./SendResume.module.scss";
import Grid from "@material-ui/core/Grid";
import {ListItem, Typography} from "@material-ui/core";
import images from "../../../assets/images";

class Resume extends Component {
    render() {
        return (
            <>
                <Grid className={Style.header} item xs={12} sm={12} md={12}>


                    <Grid item>

                        <ListItem><img src={images.userProfile.flesh1} style={{width: '30px'}} alt=""/>
                            <Typography data-test='title1'>در این قسمت شما می توانید رزومه ی خود را به صورت pdf بارگذاری کنید.</Typography></ListItem>
                        <ListItem><span data-test='title2'> توجه داشته باشید که رزومه ی شما توسط دیگر کاربران قابل مشاهده می باشد.</span></ListItem>
                    </Grid>

                    <Grid className={Style.space} item>
                        <ListItem><img data-test='flesh1' src={images.userProfile.flesh1} style={{width: '30px'}} alt=""/>
                            <span data-test='idea'>چند سایت پیشنهادی برای ساخت رزومه</span></ListItem>

                        <Grid style={{marginTop: '10px'}} item>
                            <Grid>
                                <ListItem>
                                    <img data-test='userProfile' src={images.userProfile.www} style={{width: '25px'}} alt=""/>
                                    <span><a href="https://karboom.io" target='_blank' rel={""} data-test='karboom.io'> سایت کاربوم </a></span>
                                </ListItem>
                            </Grid>
                            <Grid>
                                <ListItem>
                                    <img src={images.userProfile.www} style={{width: '25px'}} alt=""/>
                                    <span><a href="https://cvbuilder.me" target='_blank' rel={""} data-test='cvbuilder.me'> سایت سی وی بیلدر </a></span>
                                </ListItem>
                            </Grid>
                            <Grid style={{marginBottom: '50px'}}>
                                <ListItem>
                                    <img src={images.userProfile.www} style={{width: '25px'}} alt=""/>
                                    <span><a href="https://rezome.com" target='_blank' rel={""} data-test='rezome.com'> سایت رزومه دات کام </a></span>
                                </ListItem>
                            </Grid>
                        </Grid>
                    </Grid>
                </Grid>
            </>
        );
    }

}

export default Resume