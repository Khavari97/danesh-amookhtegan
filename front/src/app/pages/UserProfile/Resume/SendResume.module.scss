.resumeContainer {
  width: 90% !important;
  margin: 0 auto;
  padding: 7% 5%;
  background-color: #f5fceb;
  border: 1px solid black;
  border-radius: 5px;
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15) !important;
}

.header {
  border-bottom: 4px solid #046b00;
  justify-content: space-between;
  display: block;
  width: 100%;

  span {
    margin-right: 10px;
  }

  p {
    margin-right: 10px;
  }

  a {
    color: rgba(0, 0, 0, 0.85);
  }

  a:hover {
    color: #7DC02F;
  }

  .space {
    margin-top: 50px;

    div {
      margin-bottom: 10px;
    }
  }

}

.upload {
  width: 100%;
  margin-top: 10px;

  .content {
    margin: 30px auto;
    font-size: 1.5em;
    text-align: center;

    p {
      margin-right: 10px;
    }

    .showFile {
      margin-top: 27px;
      display: inline-flex;
      width: 100%;
      justify-content: space-between;

      .infoFile {
        margin-bottom: 20px;

        img {
          width: 40px;
        }

        a {
          font-size: 1.1rem;
          color: rgba(0, 0, 0, 0.8);
          margin-right: 10px;
          word-break: break-all;
          direction: ltr;
        }

        a:hover {
          color: #046B00;
        }

      }

      .btnDelete {
        img {
          width: 18px;

        }
      }
    }

  }

}
