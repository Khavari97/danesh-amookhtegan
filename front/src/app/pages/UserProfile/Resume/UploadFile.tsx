import React, {Component} from "react";
import Dropzone from 'react-dropzone-uploader'
import 'react-dropzone-uploader/dist/styles.css'
import LinearProgress, {LinearProgressProps} from "@material-ui/core/LinearProgress";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";
import {rootState} from "../../../redux/RootReducer";
import {connect} from "react-redux";
import {GET_RESUME} from "../../../redux/User/Profile/ActionTypes";
import Snackbar from "../../../components/material-ui/Snackbar/SnackbarUtils";
import {apiClient, defaultConfig} from '../../../api/_api'

interface IProps {
    GetResume: any
}

interface IState {
    progress: any
}

const mapStateToProps = (state: rootState) => {
    return {
        resume: state.userProfile.resume,
    }
}

const mapDispatchToProps = (dispatch: any) => {
    return {
        GetResume: (resume: string) => dispatch({type: GET_RESUME, resume: resume})
    }
}

//show progressbar
function LinearProgressWithLabel(props: LinearProgressProps & { value: number }) {
    return (
        <Box display="flex" alignItems="center">
            <Box width="100%" mr={2}>
                <LinearProgress variant="determinate" {...props} color="secondary"
                                style={{height: '10px', borderRadius: '5px'}}/>
            </Box>
            <Box minWidth={35}>
                <Typography variant="body2" color="secondary">{`${Math.round(
                    props.value,
                )}%`}</Typography>
            </Box>
        </Box>
    );
}

class UploadFile extends Component<IProps, IState> {
    constructor(props: IProps) {
        super(props)
        this.state = {progress: 0}
    }

    postData = (file: any, config: any) => {
        apiClient.post('user/profile/resume/', {
            data: file,
            headers: {...defaultConfig.headers, 'Content-Type': 'multipart/form-data'},
            onUploadProgress: (progressEvent: any) => {
                const {loaded, total} = progressEvent;
                const percent = Math.floor(loaded * 100 / total)
                if (percent < 100) {
                    this.setState({progress: percent})
                } else {
                    this.setState({progress: 100})
                }
            }
        })
            .then((response: any) => {
                this.props.GetResume(response.resume)
                Snackbar.success('فایل شما با موفقیت بارگذاری شد')
            })
            .catch((err) => {
                Snackbar.error(' متاسفانه به مشکل خوردیم')

            })
    }
    fileUpload = (file: any) => {
        const formData = new FormData();
        formData.append('resume', file)
        const config = {
            headers: {
                'content-type': 'multipart/form-data'
            }
        }
        return this.postData(formData, config)
    }

    // called every time a file's `status` changes
    handleChangeStatus = (file: any, status: any) => {
        if (status == 'done') {
            this.fileUpload(file.file)
        }
    }

    async componentDidMount() {
        apiClient.get('user/profile/resume/',
            {
                headers: {...defaultConfig.headers, 'Content-Type': 'application/pdf'},
                onUploadProgress: (progressEvent: any) => {
                    const {loaded, total} = progressEvent;
                    const percent = Math.floor(loaded * 100 / total)
                    if (percent < 100) {
                        this.setState({progress: percent})
                        console.log(percent)
                    } else {
                        this.setState({progress: 100})
                    }
                }
            })
            .then(async (response) => {
                this.props.GetResume(response.resume)
            })
            .catch((err) => {
            })
    }

    Progress = () => {
        return (
            <div style={{width: '92%', marginTop: '30px'}}>
                <LinearProgressWithLabel value={this.state.progress}/>
            </div>
        )

    }

    render() {
        return (
            <form encType='multipart/form-data'>
                <Dropzone
                    onChangeStatus={this.handleChangeStatus}
                    PreviewComponent={this.Progress}
                    maxFiles={1}
                    accept=".pdf"
                    styles={
                        {
                            dropzone: {
                                border: 'dashed 3px rgba(0,0,0,0.5)',
                                color: 'black',
                                borderRadius: '10px',
                                overflow: 'auto'

                            },
                            inputLabel: {
                                color: 'rgba(0,0,0,0.5)',
                                fontSize: '0.9rem',
                                fontFamily: 'shabnam'
                            },
                            inputLabelWithFiles: {
                                background: '#7DC02F',
                                color: 'white',
                                padding: '7px 20px',
                                borderRadius: '4px',
                                fontWeight: 300,
                                margin: '20px auto',

                            },
                            submitButton: {
                                display: 'none'

                            },
                            dropzoneActive: {
                                background: '#FFFFFF',
                                borderColor: '#046B00',
                                color: '#046B00'
                            },

                        }}
                    inputContent={'فایل مورد نظر خود را بکشید و دراین قسمت رها کنید '}
                    inputWithFilesContent={'انتخاب فایل'}
                    submitButtonContent={' آپلود فایل '}
                />
            </form>
        )
    }

}

export default connect(mapStateToProps, mapDispatchToProps)(UploadFile)