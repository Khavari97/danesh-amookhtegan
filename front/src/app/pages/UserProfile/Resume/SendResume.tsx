import React, {Component} from 'react';
import Style from "./SendResume.module.scss";
import images from "../../../assets/images";
import UploadFile from "./UploadFile";
import Resume from "./Resume";
import {Button, Grid, Icon, ListItem} from "@material-ui/core";
import {connect} from "react-redux";
import {rootState} from "../../../redux/RootReducer";
import {apiClient} from '../../../api/_api'
import Snackbar from "../../../components/material-ui/Snackbar/SnackbarUtils";
import {GET_RESUME} from "../../../redux/User/Profile/ActionTypes";
import {Helmet} from 'react-helmet'
import Popover from './../../../components/customs/Popover/Popover'
import enums from './../enums'

interface IProps {
    resume: string | null,
    GetResume: any
}

const mapStateToProps = (state: rootState) => {
    return {
        resume: state.userProfile.resume,
    }
}

const mapDispatchToProps = (dispatch: any) => {
    return {
        GetResume: (resume: string) => dispatch({type: GET_RESUME, resume: resume})
    }
}

class SendResume extends Component<IProps> {


    deleteResume = () => {
        apiClient.delete('user/profile/resume/').then(res => {
            Snackbar.success('فایل شما حذف گردید')
            this.props.GetResume(null)
        })
    }

    render() {
        const deleteIcon = (
            <Icon>
                <img data-test='userProfile' style={{width: '18px'}} alt="" src={images.userProfile.trash}/>
            </Icon>
        );

        return (
            <>
                <Helmet data-test='title'>
                    <title data-test='title'>رزومه ی من</title>
                </Helmet>
                <div className={Style.resumeContainer}>
                    <Grid container>
                        <Resume/>
                        <Grid container item className={Style.upload}>
                            <Grid className={Style.content} item container>
                                <Grid style={{display: 'table', margin: '0 auto'}} item>
                                    <ListItem>
                                        <img data-test='upload' src={images.userProfile.uploadFile} style={{width: '30px'}} alt=""/>
                                        <p data-test='uploadFile'> بارگذاری فایل رزومه </p>
                                    </ListItem>
                                </Grid>
                                <Grid container style={{display: 'table'}}>
                                    {(this.props.resume) ?
                                        <Grid item>
                                            <Grid className={Style.showFile} xs={12} item container>
                                                <Grid item xs={12} sm={6} className={Style.infoFile}>
                                                    <ListItem style={{padding: '0'}} data-test='pdf'><img
                                                        src={images.userProfile.pdfFile} alt="" data-test='pdf'/>
                                                        <a href={this.props.resume}
                                                           target="_blank">{this.props.resume.split('resume/')[1]}</a></ListItem>
                                                </Grid>
                                                <Grid item xs={12} sm={6} style={{textAlign: 'left'}}>
                                                    <Popover title={enums.popover.delete_resume}>
                                                        <Button size='large' startIcon={deleteIcon}
                                                                onClick={this.deleteResume} color="primary"
                                                                variant="contained" className={Style.btnDelete}
                                                                data-test='button'>
                                                            حذف </Button>
                                                    </Popover>
                                                </Grid>
                                            </Grid>
                                        </Grid>
                                        :
                                        <div style={{width: '100%', marginTop: '30px'}}>
                                            <UploadFile/>
                                        </div>
                                    }
                                </Grid>
                            </Grid>
                        </Grid>
                    </Grid>
                </div>
            </>
        )
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(SendResume)

