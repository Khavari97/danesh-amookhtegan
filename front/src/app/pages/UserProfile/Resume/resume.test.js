import React from 'react';
import Enzyme, { shallow} from "enzyme";
import EnzymeAdapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import Resume from "./Resume";

Enzyme.configure({ adapter: new EnzymeAdapter() });
const setup =()=> {
    return shallow(<Resume/>);
}

const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-test="${val}"]`);
};

describe('test Resume component',()=>{

    test('render without error',()=>{
        const wrapper = setup();
        expect(wrapper).toBeTruthy();
        expect(wrapper.length).toEqual(1);
    })

    test('test title1',()=>{
        const wrapper = setup().render();
        let field = findByTestAttr(wrapper, 'title1')
        expect(field.text()).toBe('در این قسمت شما می توانید رزومه ی خود را به صورت pdf بارگذاری کنید.')
        expect(field.length).toEqual(1);
    })

    test('test title2',()=>{
        const wrapper = setup().render();
        let field = findByTestAttr(wrapper, 'title2')
        expect(field.text()).toBe(' توجه داشته باشید که رزومه ی شما توسط دیگر کاربران قابل مشاهده می باشد.')
        expect(field.length).toEqual(1);
    })

    test('test image src & links',()=>{
        const wrapper = setup().render();
        expect(findByTestAttr(wrapper,'flesh1').prop('src')).toBe('flesh1.svg');
        expect(findByTestAttr(wrapper,'idea').text()).toBe('چند سایت پیشنهادی برای ساخت رزومه');
        expect(findByTestAttr(wrapper,'userProfile').prop('src')).toBe('www.svg');
        expect(findByTestAttr(wrapper,'karboom.io').prop('href')).toBe('https://karboom.io');
        expect(findByTestAttr(wrapper,'karboom.io').text()).toBe(' سایت کاربوم ');
        expect(findByTestAttr(wrapper,'cvbuilder.me').prop('href')).toBe('https://cvbuilder.me');
        expect(findByTestAttr(wrapper,'cvbuilder.me').text()).toBe(' سایت سی وی بیلدر ');
        expect(findByTestAttr(wrapper,'rezome.com').prop('href')).toBe('https://rezome.com');
        expect(findByTestAttr(wrapper,'rezome.com').text()).toBe(' سایت رزومه دات کام ');
    })

    test('snapshot',()=>{
        const shot = <Resume/>
        const snapshot = renderer.create(shot).toJSON();
        expect(snapshot).toMatchSnapshot();

    })
})