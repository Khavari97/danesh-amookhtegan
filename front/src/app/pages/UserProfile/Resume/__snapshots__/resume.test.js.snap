// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Resume component snapshot 1`] = `
<div
  className="MuiGrid-root header MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <li
      className="MuiListItem-root MuiListItem-gutters"
      disabled={false}
    >
      <img
        alt=""
        src="flesh1.svg"
        style={
          Object {
            "width": "30px",
          }
        }
      />
      <p
        className="MuiTypography-root MuiTypography-body1"
        data-test="title1"
      >
        در این قسمت شما می توانید رزومه ی خود را به صورت pdf بارگذاری کنید.
      </p>
    </li>
    <li
      className="MuiListItem-root MuiListItem-gutters"
      disabled={false}
    >
      <span
        data-test="title2"
      >
         توجه داشته باشید که رزومه ی شما توسط دیگر کاربران قابل مشاهده می باشد.
      </span>
    </li>
  </div>
  <div
    className="MuiGrid-root space MuiGrid-item"
  >
    <li
      className="MuiListItem-root MuiListItem-gutters"
      disabled={false}
    >
      <img
        alt=""
        data-test="flesh1"
        src="flesh1.svg"
        style={
          Object {
            "width": "30px",
          }
        }
      />
      <span
        data-test="idea"
      >
        چند سایت پیشنهادی برای ساخت رزومه
      </span>
    </li>
    <div
      className="MuiGrid-root MuiGrid-item"
      style={
        Object {
          "marginTop": "10px",
        }
      }
    >
      <div
        className="MuiGrid-root"
      >
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
        >
          <img
            alt=""
            data-test="userProfile"
            src="www.svg"
            style={
              Object {
                "width": "25px",
              }
            }
          />
          <span>
            <a
              data-test="karboom.io"
              href="https://karboom.io"
              rel=""
              target="_blank"
            >
               سایت کاربوم 
            </a>
          </span>
        </li>
      </div>
      <div
        className="MuiGrid-root"
      >
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
        >
          <img
            alt=""
            src="www.svg"
            style={
              Object {
                "width": "25px",
              }
            }
          />
          <span>
            <a
              data-test="cvbuilder.me"
              href="https://cvbuilder.me"
              rel=""
              target="_blank"
            >
               سایت سی وی بیلدر 
            </a>
          </span>
        </li>
      </div>
      <div
        className="MuiGrid-root"
        style={
          Object {
            "marginBottom": "50px",
          }
        }
      >
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
        >
          <img
            alt=""
            src="www.svg"
            style={
              Object {
                "width": "25px",
              }
            }
          />
          <span>
            <a
              data-test="rezome.com"
              href="https://rezome.com"
              rel=""
              target="_blank"
            >
               سایت رزومه دات کام 
            </a>
          </span>
        </li>
      </div>
    </div>
  </div>
</div>
`;
